File: string/shuffle-string.js
// https://leetcode.com/problems/shuffle-string/
// tags - string
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    const chars = Array(s.length);
    for (let i = 0; i < s.length; i++) {
        chars[indices[i]] = s[i];
    }
    
    return chars.join('');
    
};

File: string/shuffle-string.py
# https://leetcode.com/problems/shuffle-string/
# tags - string
class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        chars = list(s)
        for i, j in zip(range(len(s)), indices):
            chars[j] = s[i]
        return ''.join(chars)